apis/python/src/tiledbsoma/_fastercsx.py

```diff
@@ -0,0 +1,269 @@
+        data: NDArrayNumber,
+        shape: Tuple[int, int],
+        format: Literal["csc", "csr"],
+        sorted: bool,
+        context: SOMATileDBContext,
+    ) -> None:
+        """Construct from PJV format. Not intended for direct use - see instead
+        static factory methods `from_ijd` and `from_pjv`.
+        """
+        assert len(data) == len(indices)
+        assert len(data) <= np.iinfo(indptr.dtype).max
+        assert shape[1] <= np.iinfo(indices.dtype).max
+        assert indptr[-1] == len(data) and indptr[0] == 0
+
+        self.shape = shape
+        self.indptr = indptr
+        self.indices = indices
+        self.data = data
+        self.format = format
+        self.sorted = sorted
+        self.context = context
+
+    @staticmethod
+    def from_ijd(
+        i: NDArrayIndex | Sequence[NDArrayIndex],
+        j: NDArrayIndex | Sequence[NDArrayIndex],
+        d: NDArrayNumber | Sequence[NDArrayNumber],
+        shape: Tuple[int, int],
+        format: Literal["csc", "csr"],
+        make_sorted: bool,
+        context: SOMATileDBContext,
+    ) -> CompressedMatrix:
+        """Factory method accepting COO points stored in IJD vectors."""
+        i = i if isinstance(i, collections.abc.Sequence) else (i,)
+        j = j if isinstance(j, collections.abc.Sequence) else (j,)
+        d = d if isinstance(d, collections.abc.Sequence) else (d,)
+
+        if format == "csc":
+            i, j = j, i
+            n_major, n_minor = shape[1], shape[0]
+        else:
+            n_major, n_minor = shape[0], shape[1]
+
+        nnz = sum(len(d_) for d_ in d)
+        index_dtype = CompressedMatrix._smallest_index_dtype(nnz)
+        indptr = np.zeros((n_major + 1), dtype=index_dtype)
+        indices = np.empty((nnz,), dtype=index_dtype)
+        data = np.empty((nnz,), dtype=d[0].dtype)
+        compress_coo(
+            context.native_context, (n_major, n_minor), i, j, d, indptr, indices, data
+        )
+        if make_sorted:
+            sort_indices(context.native_context, indptr, indices, data)
+        return CompressedMatrix(
+            indptr, indices, data, shape, format, make_sorted, context
+        )
+
+    @staticmethod
+    def from_pjd(
+        p: NDArrayIndex,
+        j: NDArrayIndex,
+        d: NDArrayNumber,
+        shape: Tuple[int, int],
+        format: Literal["csc", "csr"],
+        make_sorted: bool,
+        context: SOMATileDBContext,
+    ) -> CompressedMatrix:
+        """Factory method accepting pointers, indices and data vectors."""
+        return CompressedMatrix(p, j, d, shape, format, make_sorted, context)
+
+    @staticmethod
+    def from_soma(
+        tables: pa.Table | Sequence[pa.Table],
+        shape: Tuple[int, int],
+        format: Literal["csc", "csr"],
+        make_sorted: bool,
+        context: SOMATileDBContext,
+    ) -> CompressedMatrix:
+        """Factory method accepting a sequence of Arrow tables containing SOMA sparse matrix data.
```

Comment:
So, `from_arrow` maybe? Just a thought.

Reply:
No, I think that is not correct.  It is a SOMA data structure (due to column names, assumption about column types, etc), encoded as an Arrow table.  You can't hand random Arrow data to it and get anything useful.

I think it has the right name.

Reply:
Cool, thanks!

Reply:
I guess we could go super verbose, e,g, from_soma_coo_data_as_arrow_table (or a more elegant variant on that), but I think I'd rather just make sure the docstring is clear :-)

Reply:
Works for me!